modules/@apostrophecms/doc-type/index.js

```diff
@@ -97,12 +97,27 @@ module.exports = {
           self.apos.schema.prepareForStorage(req, doc);
         },
         slugPrefix(req, doc) {
-          if (self.options.slugPrefix) {
+          const prefix = self.options.slugPrefix;
+          if (prefix) {
             if (!doc.slug) {
               doc.slug = 'none';
             }
-            if (!doc.slug.startsWith(self.options.slugPrefix)) {
-              doc.slug = `${self.options.slugPrefix}${doc.slug}`;
+
+            let archivePrefix;
+            const archivedRegexp = new RegExp(`^deduplicate-[a-z0-9]+-${prefix}`);
```

Comment:
Always use `self.apos.util.regexpQuote` on a string literal before treating it as part of a regular expression.